crates/bevy_render/src/lib.rs

```diff
@@ -102,13 +108,32 @@ use crate::{
 };
 use alloc::sync::Arc;
 use bevy_app::{App, AppLabel, Plugin, SubApp};
-use bevy_asset::{load_internal_asset, weak_handle, AssetApp, AssetServer, Handle};
+use bevy_asset::{AssetApp, AssetServer};
 use bevy_ecs::{prelude::*, schedule::ScheduleLabel};
 use bitflags::bitflags;
 use core::ops::{Deref, DerefMut};
 use std::sync::Mutex;
 use tracing::debug;
 
+// TODO: this should be replaced by proper asset handling in the shader loader.
+/// Inline shader as an `embedded_asset` and load it permanently.
+///
+/// This works around a limitation of the shader loader not properly loading
+/// dependencies of shaders.
+#[macro_export]
+macro_rules! load_and_forget_shader {
```

Comment:
I wonder if we should call this `load_and_forget_shader`. It certainly is descriptive, but the fact that we are forgetting is kind of an implementation detail. As in, what if instead of forgetting we store the handle in some resource that stores all our shader handles? (I think for the time being forget is the best impl though)

Could we just name this something like `load_shader_library`? That doesn't really explain what "loading" means, but that's ok - whatever it is, it will be a usable shader.

Reply:
sure